### PR TITLE
Adds AndroidSchedulers.currentThread()

### PR DESCRIPTION
**Pull request by manidesto**

### CODE REVIEW

File: rxandroid/src/main/java/rx/android/schedulers/AndroidSchedulers.java [hunk 1]
```diff
@@ -38,4 +38,10 @@ public static Scheduler mainThread() {
                 RxAndroidPlugins.getInstance().getSchedulersHook().getMainThreadScheduler();
         return scheduler != null ? scheduler : MainThreadSchedulerHolder.MAIN_THREAD_SCHEDULER;
     }
+
+    public static Scheduler currentThread() {
```
Comment (artem-zinnatullin) [inline rxandroid/src/main/java/rx/android/schedulers/AndroidSchedulers.java]: Needs javadoc

Comment (artem-zinnatullin) [inline rxandroid/src/main/java/rx/android/schedulers/AndroidSchedulers.java]: And at least two more tests: 

1. For a `Thread` with `Looper` to check that it picks its looper and runs the code on that thread.
2. For a `Thread` without `Looper` to see how it fails with `Can't create handler inside thread that has not called Looper.prepare()`.

**Comment by manidesto** [comment]
@artem-zinnatullin 
Not sure how to test that. Any suggestions?

**Comment by dlew** [comment]
I'd like to see where this becomes important in practice.

It sort of seems like something you could do without a custom `Scheduler`, e.g. by querying the current `Looper` before calling `subscribeOn` or `observeOn`

**Comment by JakeWharton** [comment]
I ask in the issue for this feature for some use cases which was not
fulfilled.

**Comment by manidesto** [comment]
To return an observable that does work in a background thread but emits items in the same thread the consumer of the api called the function. 
Using Scheduler.immediate() or Scheduler.trampoline() may break the app if the api is called from the main thread.

Wasn't able to unit test my presenter due to issue #238 . So when moving the `observeOn()` part into the API, wanted this function so that I can call my API from any thread. Will not be much of a need if the fix for #238 is released I guess.

**Comment by artem-zinnatullin** [comment]
So, if it's just for testing (I don't see any significant use cases for real usage at the moment), you should be fine with fix #238.

I'd just suggest you to make schedulers configurable in your Presenters and pass `Schedulers.trampoline()` or `Schedulers.immediate()` for tests.

@manidesto let's close this then?

**Comment by manidesto** [comment]
Yeah. But please ready the next release as soon as possible.

**Pull request closed without merging**